libraries/AP_BattMonitor/AP_BattMonitor.cpp

```diff
@@ -113,13 +114,27 @@ AP_BattMonitor::init()
                 drivers[instance] = new AP_BattMonitor_Analog(*this, state[instance], _params[instance]);
                 break;
             case AP_BattMonitor_Params::BattMonitor_TYPE_SOLO:
+                _params[instance]._bus.set_default(AP_BATTMONITOR_SMBUS_BUS_INTERNAL),
                 drivers[instance] = new AP_BattMonitor_SMBus_Solo(*this, state[instance], _params[instance],
-                                                                  hal.i2c_mgr->get_device(AP_BATTMONITOR_SMBUS_BUS_INTERNAL, AP_BATTMONITOR_SMBUS_I2C_ADDR,
+                                                                  hal.i2c_mgr->get_device(_params[instance]._bus, AP_BATTMONITOR_SMBUS_I2C_ADDR,
                                                                                           100000, true, 20));
                 break;
+            case AP_BattMonitor_Params::BattMonitor_TYPE_Xeno:
+                _params[instance]._bus.set_default(AP_BATTMONITOR_SMBUS_BUS_INTERNAL),
+                drivers[instance] = new AP_BattMonitor_WithCellCount(*this, state[instance], _params[instance],
+                                                                  hal.i2c_mgr->get_device(_params[instance]._bus, AP_BATTMONITOR_SMBUS_I2C_ADDR,
+                                                                                          100000, true, 20), 3);
+                break;
+            case AP_BattMonitor_Params::BattMonitor_TYPE_Endurance:
+                _params[instance]._bus.set_default(AP_BATTMONITOR_SMBUS_BUS_INTERNAL),
+                drivers[instance] = new AP_BattMonitor_WithCellCount(*this, state[instance], _params[instance],
+                                                                     hal.i2c_mgr->get_device(_params[instance]._bus, AP_BATTMONITOR_SMBUS_I2C_ADDR,
+                                                                                             100000, true, 20), 6);
+                break;
             case AP_BattMonitor_Params::BattMonitor_TYPE_MAXELL:
+                _params[instance]._bus.set_default(AP_BATTMONITOR_SMBUS_BUS_EXTERNAL),
```

Comment:
This is really messy, I get that the maxell driver wanted the external by default, but this sorta magic twiddling is a bit confusing, and can lead to people accidentally. I think a default param to scan all buses would probably be less painful at that point.

Reply:
Can lead to people accidentally...what?